packages/ui/src/stores/player.js

```diff
@@ -98,20 +102,34 @@ export const useStore = defineStore('player', {
       const tokenInfo = this.getToken()
       const request = await this.api.trade({
         token: tokenInfo.token,
-        to: key,
-        bufficorn: this.selectedBufficorn
+        to: key
       })
       if (request.error) {
         this.setError('trade', request.error)
-        router.push('/')
+        router.push('/init-game')
       } else {
         this.clearError('trade')
         this.tradeInfo = request
-        this.getInfo()
-        router.push('/')
+        router.push('/init-game')
+      }
+    },
+    async updateSelectedBufficorn (index) {
+      const tokenInfo = this.getToken()
+      const request = await this.api.selectBufficorn({
+        token: tokenInfo.token,
+        bufficorn: index
+      })
+      if (request.error) {
+        this.setError('updateSelectedBufficorn', request.error)
+        router.push('/init-game')
+      } else {
+        this.clearError('updateSelectedBufficorn')
+        this.selectedBufficorn = request
+        await this.getInfo()
```

Comment:
We can call `selectBufficorn` only when the index is different than the already selected one

```suggestion
      if (index !== this.selectedBufficorn) {
        const tokenInfo = this.getToken()
        const request = await this.api.selectBufficorn({
          token: tokenInfo.token,
          bufficorn: index
        })
        if (request.error) {
          this.setError('updateSelectedBufficorn', request.error)
          router.push('/init-game')
        } else {
          this.clearError('updateSelectedBufficorn')
          this.selectedBufficorn = request
          await this.getInfo()
        }
```